src/graph_notebook/magics/streams.py

```diff
@@ -0,0 +1,225 @@
+            (records, first_event, last_event) = self.stream_client.get_events(language, event_id, STREAM_AT)
+            self.update_slider_min_max_values(language)
+            self.slider.value = first_event.commit_num
+            self.show_records(records)
+            self.first_displayed_event_id.update(first_event)
+            self.last_displayed_event_id.update(last_event)
+            
+    def init_display(self, language):
+        self.history = queue.LifoQueue(100)
+        self.back_button.disabled = True
+        self.update_slider_min_max_values(language)
+        self.slider.value = self.slider.min
+        (records, first_event, last_event) = self.stream_client.get_events(language, EventId(self.slider.min, 1), STREAM_AT)
+        self.show_records(records)
+        self.first_displayed_event_id.update(first_event)
+        self.last_displayed_event_id.update(last_event)
+       
+
+    def show_records(self, records):
+        
+        html = '''<html><body><table style="border: 1px solid black">'''
+        
+        html += '''<tr>
+                   <th style="text-align: center" >Tx/Op#</th>
+                   <th style="text-align: center">Operation</th>
+                   <th style="text-align: center">Data</th>
+                   </tr>'''
+            
+        commit_num = None
+     
+        for record in records:
+            current_commit_num = record['eventId']['commitNum']
+            
+            data = json.dumps(record['data']).replace('&', '&amp;').replace('<', '&lt;')
+            
+            if commit_num is None or current_commit_num != commit_num:
+                commit_num = current_commit_num
+                html += '<tr title="The commit number for this transaction" style="border: 1px solid black; background-color: gainsboro ; font-weight: bold;">'
+                html += '<td style="border: 1px solid black; vertical-align: top; text-align: left;" colspan="3">{}</td>'.format(commit_num)
+                html += '</tr><tr style="border: 1px solid black;">'     
+            
+            html += '<tr  title="The operation number within this transaction" style="border: 1px solid black; background-color: white;">'
+            html += '''<td style="border: 1px solid black; vertical-align: top;">{}</td>
+            <td style="border: 1px solid black; vertical-align: top;">{}</td>
+            <td style="border: 1px solid black; vertical-align: top; text-align: left;">{}</td></tr>'''.format(
+                record['eventId']['opNum'], 
+                record['op'],
+                data)
+           
+        html += '</table></body></html>'
+        
+        self.out.clear_output(wait=True)
+        with self.out:
+            display(HTML(html))
+            
+    def update_slider_min_max_values(self, language):
+        
+        new_min = self.stream_client.get_first_commit_num(language)
+        new_max = self.stream_client.get_last_commit_num(language)
```

Comment:
One thing I noticed while testing, this line does not fail gracefully if the connection to the Neptune instance either does not exist upon running `%stream_viewer`, or it expires while the viewer is running and we try to scroll the stream. 

We should catch the initial exception (`from requests.exceptions import ConnectionError`) and output a more concise error message, something like:
```
try:
    new_max = self.stream_client.get_last_commit_num(language)
except ConnectionError as e:
    print("Unable to reach Neptune, please check your host configuration and/or connection.")
    print(f"Full error: {e}")
    return
```